Tools/BuildSrc/build_packages.py

```diff
@@ -0,0 +1,277 @@
+        err = True
+
+    return err
+
+
+def _create_packages(alusus_build_packages, alusus_install_location, alusus_packages_location, verbose_output=False):
+    to_be_packaged_dir = _create_to_be_packaged_dir(
+        alusus_install_location, alusus_packages_location)
+    alusus_packages_location = os.path.join(
+        alusus_packages_location, "AlususPackages")
+    common.remove_path(alusus_packages_location, follow_symlinks=False)
+    os.makedirs(alusus_packages_location, exist_ok=True)
+
+    version_info = VersionInfo(
+        common.ALUSUS_REPO_PATH, verbose_output=verbose_output)
+    package_basename = sanitize_filename.sanitize("alusus-{version}{revision}-{machine}-{platform}{abi}".format(
+        version=version_info.version_lossy,
+        revision="-" + version_info.revision_lossy if version_info.revision_lossy else "",
+        machine=alusus_build_packages.value.alusus_target_triplet.value.machine,
+        platform=alusus_build_packages.value.alusus_target_triplet.value.platform,
+        abi="-" + alusus_build_packages.value.alusus_target_triplet.value.abi if alusus_build_packages.value.alusus_target_triplet.value.abi else ""
+    ))
+
+    err = alusus_build_packages.value.create_packages_func(
+        alusus_packages_location, to_be_packaged_dir, package_basename, version_info, alusus_build_packages.value.alusus_target_triplet.value.machine,
+        verbose_output)
+
+    msg.info_msg("Cleaning up to-be-packaged directory...")
+    shutil.rmtree(to_be_packaged_dir, ignore_errors=True)
+
+    if err:
+        msg.warn_msg("Some optional packages failed to build.")
+    else:
+        msg.success_msg("Building packages.")
+
+
+class _BuildPackagesWithPackageBuildingFunc:
+    def __init__(self, alusus_target_triplet, create_packages_func) -> None:
+        self._alusus_target_triplet = alusus_target_triplet
+        self._create_packages_func = create_packages_func
+
+    @property
+    def alusus_target_triplet(self):
+        return self._alusus_target_triplet
+
+    @property
+    def create_packages_func(self):
+        return self._create_packages_func
+
+
+class BuildPackages(common.ExtendedEnum):
+    @staticmethod
+    def from_alusus_target_triplet(alusus_target_triplet: TargetTriplet):
+        for item in BuildPackages.list():
+            if item.value.alusus_target_triplet == alusus_target_triplet:
+                return item
+        raise NotImplementedError(
+            "{alusus_target_triplet_class_name} has no matching package building function.".format(
+                alusus_target_triplet_class_name=json.dumps(
+                    alusus_target_triplet.__class__.__name__)
+            ))
+
+    create_packages = _create_packages
+
+    X64_LINUX = _BuildPackagesWithPackageBuildingFunc(
+        TargetTriplet.X64_LINUX, _create_packages_linux)
+    X64_OSX = _BuildPackagesWithPackageBuildingFunc(
+        TargetTriplet.X64_OSX, _create_packages_osx)
```

Comment:
I'm having a hard time understanding this code. Is this using metaprogramming to build the class before running it?

Reply:
The purpose of the file is very straight forward so I'm guessing it's possible to simplify it by avoiding any Python magic and simply doing straight forward functions, i.e. a function that receives the build parameters and routes the code with simple if statements.